### DISCUSSION TITLE
Drag and Drop

### DISCUSSION THREAD

**Discussion by zsviczian** [excalidraw/excalidraw#3819] [Q&A]
I am trying to implement drag and drop to allow adding links into text elements by dragging files from Obsidian File explorer to Excalidraw canvas.  (i.e. dropping a file on the canvas would create a new text element with a link to the file e.g. [[link to file]])
To achieve this, I implemented `onDragOver`, `onDragLeave` and `onDrop`  for my excalidraw-wrapper DIV.
I successfully capture the drag event, but after it completes I get this error, seemingly from the Excalidraw component. How can I suppress this?

![image](https://user-images.githubusercontent.com/14358394/125174980-7571cb00-e1c9-11eb-898e-e78ebec844d6.png)

**Comment by dwelle** [comment]
Yeah, we'll need to figure out how to differentiate excalidraw JSON drag-n-dropped onto canvas as opposed to everything else.
